File: src/IsbInterop.CSharp/Data/Wrappers/Requisite.cs
using IsbInterop.DataTypes.Enumerable;
using System;

namespace IsbInterop.Data.Wrappers
{
  /// <summary>
  /// Обертка над IRequisite.
  /// </summary>
  internal class Requisite : IsbComObjectWrapper, IRequisite
  {
    #region Поля и свойства

    /// <summary>
    /// Значение реквизита в качестве даты.
    /// </summary>
    public DateTime AsDate
    {
      get { return (DateTime)this.GetRcwProperty("AsDate"); }
    }

    /// <summary>
    /// Целочисленное значение реквизита.
    /// </summary>
    public int AsInteger
    {
      get { return (int)this.GetRcwProperty("AsInteger"); }
    }

    /// <summary>
    /// Строковое значение реквизита.
    /// </summary>
    public string AsString
    {
      get { return (string)this.GetRcwProperty("AsString"); }
    }

    /// <summary>
    /// Тип реквизита.
    /// </summary>
    public TReqDataType DataType
    {
      get
      {
        int dataType = (int)this.GetRcwProperty("DataType");

        if (Enum.IsDefined(typeof(TReqDataType), dataType))
          return (TReqDataType)dataType;
        else
          return TReqDataType.rdtUnknown;
      }
    }

    /// <summary>
    /// Имя поля реквизита в базе.
    /// </summary>
    public string FieldName
    {
      get { return (string)this.GetRcwProperty("FieldName"); }
    }

    /// <summary>
    /// Признак, указывающий, что реквизит равен null.
    /// </summary>
    public bool IsNull
    {
      get { return (bool)this.GetRcwProperty("IsNull"); }
    }

    /// <summary>
    /// Имя реквизита.
    /// </summary>
    public string Name
    {
      get { return (string)this.GetRcwProperty("Name"); }
    }

    /// <summary>
    /// Значение реквизита.
    /// </summary>
    public object Value
    {
      get { return this.GetRcwProperty("Value"); }
      set { this.SetRcwProperty("Value", value); }
    }

    #endregion

    #region Конструктор

    /// <summary>
    /// Конструктор.
    /// </summary>
    /// <param name="rcwRequisite">COM-объект реквизит.</param>
    internal Requisite(object rcwRequisite) : base(rcwRequisite) { }

    #endregion
  }
}

File: src/IsbInterop/Data/Wrappers/Requisite.vb
Imports IsbInterop.DataTypes.Enumerable

Namespace Data.Wrappers
    ''' <summary>
    ''' Обертка над IRequisite.
    ''' </summary>
    Friend Class Requisite
        Inherits IsbComObjectWrapper
        Implements IRequisite

#Region "Поля и свойства"

        ''' <summary>
        ''' Значение реквизита в качестве даты.
        ''' </summary>
        Public ReadOnly Property AsDate() As DateTime Implements IRequisite.AsDate
            Get
                Return DirectCast(Me.GetRcwProperty("AsDate"), DateTime)
            End Get
        End Property

        ''' <summary>
        ''' Целочисленное значение реквизита.
        ''' </summary>
        Public ReadOnly Property AsInteger() As Integer Implements IRequisite.AsInteger
            Get
                Return CInt(Me.GetRcwProperty("AsInteger"))
            End Get
        End Property

        ''' <summary>
        ''' Строковое значение реквизита.
        ''' </summary>
        Public ReadOnly Property AsString() As String Implements IRequisite.AsString
            Get
                Return DirectCast(Me.GetRcwProperty("AsString"), String)
            End Get
        End Property

        ''' <summary>
        ''' Тип реквизита.
        ''' </summary>
        Public ReadOnly Property DataType() As TReqDataType Implements IRequisite.DataType
            Get
                Dim requisiteDataType As Integer = CInt(Me.GetRcwProperty("DataType"))

                If [Enum].IsDefined(GetType(TReqDataType), requisiteDataType) Then
                    Return DirectCast(requisiteDataType, TReqDataType)
                Else
                    Return TReqDataType.rdtUnknown
                End If
            End Get
        End Property

        ''' <summary>
        ''' Имя поля реквизита в базе.
        ''' </summary>
        Public ReadOnly Property FieldName() As String Implements IRequisite.FieldName
            Get
                Return DirectCast(Me.GetRcwProperty("FieldName"), String)
            End Get
        End Property

        ''' <summary>
        ''' Признак, указывающий, что реквизит равен null.
        ''' </summary>
        Public ReadOnly Property IsNull() As Boolean Implements IRequisite.IsNull
            Get
                Return CBool(Me.GetRcwProperty("IsNull"))
            End Get
        End Property

        ''' <summary>
        ''' Имя реквизита.
        ''' </summary>
        Public ReadOnly Property Name() As String Implements IRequisite.Name
            Get
                Return DirectCast(Me.GetRcwProperty("Name"), String)
            End Get
        End Property

        ''' <summary>
        ''' Значение реквизита.
        ''' </summary>
        Public Property Value() As Object Implements IRequisite.Value
            Get
                Return Me.GetRcwProperty("Value")
            End Get
            Set(requisiteValue As Object)
                Me.SetRcwProperty("Value", requisiteValue)
            End Set
        End Property

#End Region

#Region "Конструктор"

        ''' <summary>
        ''' Конструктор.
        ''' </summary>
        ''' <param name="rcwRequisite">COM-объект реквизит.</param>
        Friend Sub New(rcwRequisite As Object)
            MyBase.New(rcwRequisite)
        End Sub

#End Region

    End Class
End Namespace